modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridAbstractCommunicationClient.java

```diff
@@ -32,19 +39,27 @@
     /** Reservations. */
     private final AtomicBoolean closed = new AtomicBoolean();
 
-    /** Metrics listener. */
-    protected final GridNioMetricsListener metricsLsnr;
-
     /** */
     private final int connIdx;
 
+    /** Received bytes count metric. */
+    @Nullable protected final LongAdderMetric rcvdBytesCntMetric;
+
+    /** Sent bytes count metric. */
+    @Nullable protected final LongAdderMetric sentBytesCntMetric;
+
     /**
      * @param connIdx Connection index.
-     * @param metricsLsnr Metrics listener.
+     * @param mreg Metrics registry.
      */
-    protected GridAbstractCommunicationClient(int connIdx, @Nullable GridNioMetricsListener metricsLsnr) {
+    protected GridAbstractCommunicationClient(int connIdx, @Nullable MetricRegistry mreg) {
```

Comment:
Why `mreg` can be null?

Reply:
This class might be used by components other than TCP SPI. You can see that "GridNioMetricsListener" was nullable as well.